<launch>
	<arg name="uav_id" default="1"/>
    <arg name="flag_printf" default="true"/>
    <arg name="external_source" default="2"/>
	<arg name="server" default="192.168.31.163"/>

    <!-- 启动 vrpn_client_ros -->
    <include file="$(find sunray_px4)/launch/sunray_vrpn.launch" if="$(eval external_source == 0)">
        <arg name="server" value="$(arg server)"/>
        <arg name="uav_id" value="$(arg uav_id)"/>
    </include>

    <!-- 启动 vision_pose_node -->
    <node pkg="sunray_uav_control" type="vision_pose_node" name="vision_pose_node_$(arg uav_id)" output="screen">
        <param name="flag_printf" value="$(arg flag_printf)" />
        <param name="uav_id" value="$(arg uav_id)" />
        <param name="external_source" value="$(arg external_source)" />
    </node>

    <!-- Rviz config -->
    <arg name="rviz_enable" default="true"/>
    <arg name="rivz_config" default="$(find sunray_uav_control)/rviz/uav.rviz"/>
	<!-- 启动Rviz-->
	<group if="$(arg rviz_enable)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>
</launch>